File: api-server/src/infrastructure/templates/defaultTemplateProvider.ts
// Implemented for spec: agent/specs/meal-appointment-create-appointment-backend-spec.md

import type { ActiveTemplateProvider } from '../../application/appointments/activeTemplateService';

export class DefaultActiveTemplateProvider implements ActiveTemplateProvider {
  async loadActiveTemplateIds(): Promise<string[]> {
    return ['default_weekly'];
  }
}

File: api-server/internal/infrastructure/repository/default_template_provider.go
package repository

import "context"

type ActiveTemplateProvider interface {
	LoadActiveTemplateIDs(ctx context.Context) ([]string, error)
}

type DefaultActiveTemplateProvider struct{}

func NewDefaultActiveTemplateProvider() *DefaultActiveTemplateProvider {
	return &DefaultActiveTemplateProvider{}
}

func (p *DefaultActiveTemplateProvider) LoadActiveTemplateIDs(_ context.Context) ([]string, error) {
	return []string{"default_weekly"}, nil
}
